### PR TITLE
P4-2259 reworking the pricing import process for consistency purposes.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
The changes for this PR are predominantly around consistency.  The process for locations and prices is similar.  I have also renamed things where I thought it makes sense.  I have introduced wrapper classes to better encapsulate the behaviour for both the locations and pricing spreadsheet.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt [hunk 1]
```diff
@@ -12,46 +12,38 @@ import java.time.Duration
...
-            rows.forEachIndexed{ index, row ->
-                run {
-                    Result.runCatching {
-                        locationTab.map(row).let {
-                            repo.save(it)
-                            total++
-                        }
-                    }.onFailure { errors.add(LocationImportError(locationTab, index + rowOffset, it.cause?.cause ?: it)) }
-                }
+        LocationsSpreadsheet.Tab.values().map { tab ->
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt:33]: minor; but may as well be forEach given you are not mapping it to anything

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt [hunk 2]
```diff
@@ -12,46 +12,38 @@ import java.time.Duration
...
+            spreadsheet.getRowsFrom(tab).forEach { row ->
+                Result.runCatching {
+                    spreadsheet.mapToLocation(row).let {
+                        repo.save(it)
+                        total++
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt:38]: I couldn't see where you later reference `total`

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        private val row: Row = sheet.createRow(0)
+
+        @Test
+        fun `no errors by default`() {
+            assertThat(spreadsheet.errors).isEmpty()
+        }
+
+        @Test
+        fun `errors are recorded`() {
+            val exception: RuntimeException = RuntimeException("something went wrong")
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt:38]: I guess you could just write `val exception = RuntimeException("something went wrong")`

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    inner class MappingRowsPrice {
+
+        private val row: Row = sheet.createRow(1)
+
+        init {
+            row.createCell(0).setCellValue(1.0)
+            row.createCell(1).setCellValue("from site")
+            row.createCell(2).setCellValue("to site")
+            row.createCell(3).setCellValue(100.00)
+        }
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt:56]: or you could write something like 
`        
private val row: Row = sheet.createRow(1).apply {
            this.createCell(0).setCellValue(1.0)
            this.createCell(1).setCellValue("from site")
            this.createCell(2).setCellValue("to site")
            this.createCell(3).setCellValue(100.00)
        }
`

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt (hunk 2 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt:38]: Line 46

Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt:38]: I think ultimately I'd like to change this...

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt (hunk 3 above)
Comment (matt-crowson-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt]: done

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt (hunk 4 above)
Comment (matt-crowson-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt:54]: done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt (hunk 1 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt]: done

**Review by jcavell — APPROVED** [review]

**Pull request merged**